witness/golang/cmd/witness/internal/witness/witness.go

```diff
@@ -117,48 +115,93 @@ func (w *Witness) GetCheckpoint(logID string) ([]byte, error) {
 	}
 	cosigned, err := note.Sign(n, w.Signer)
```

Comment:
Not to be addressed in this PR, but I wonder if you considered co-signing the checkpoint when performing `Update`, so that `GetCheckpoint` would return it straight away with a smaller overhead.

Reply:
I did talk to @AlCutter about it!  It is almost certainly better - at some point I thought that there was a reason to keep them separate but now I don't remember that reason so will likely go ahead and implement it at some point :).